service-clients-core/src/main/java/com/flipkart/poseidon/serviceclients/FutureTaskResultToDomainObjectPromiseWrapper.java

```diff
@@ -35,13 +36,24 @@
 public class FutureTaskResultToDomainObjectPromiseWrapper<DomainObject> implements Promise<DomainObject> {
 
     private final Future<TaskResult> future;
+    private final List<Future<TaskResult>> futureList = new ArrayList<>();
     private final List<PromiseListener> promiseListeners = new ArrayList<>();
     private PromiseBrokenException promiseBrokenException;
+    private ResponseMerger<DomainObject> responseMerger;
 
     public FutureTaskResultToDomainObjectPromiseWrapper(Future<TaskResult> future) {
         this.future = future;
     }
 
+    public FutureTaskResultToDomainObjectPromiseWrapper(ResponseMerger<DomainObject> responseMerger){
+        this.future = null;
+        this.responseMerger = responseMerger;
+    }
+
+    public FutureTaskResultToDomainObjectPromiseWrapper() {
+        future = null;
+    }
+
```

Comment:
Empty constructor is removed. 
Since future is declared as final, should be initialized in constructor.